File: organic_reactions_project/list_models.py
#!/usr/bin/env python3

import os
from google import genai

def list_available_models():
    client = genai.Client(api_key=os.getenv("GEMINI_API_KEY"))

    try:
        models = client.models.list()
        print("Available Gemini models:")
        for model in models:
            print(f"  - {model.name}")
    except Exception as e:
        print(f"Error listing models: {e}")

if __name__ == "__main__":
    list_available_models()

File: chem_canvas/chem_canvas/list_models.ts

import { GoogleGenAI } from '@google/genai';

const apiKey = process.env.GEMINI_API_KEY || 'YOUR_API_KEY';
const genAI = new GoogleGenAI({ apiKey });

async function listModels() {
    try {
        const response = await genAI.models.list();
        console.log('Available Models:');
        for (const model of response.models) {
            console.log(`- ${model.name}`);
            console.log(`  Supported Generation Methods: ${model.supportedGenerationMethods}`);
            console.log(`  Input Token Limit: ${model.inputTokenLimit}`);
            console.log(`  Output Token Limit: ${model.outputTokenLimit}`);
        }
    } catch (error) {
        console.error('Error listing models:', error);
    }
}

listModels();
